### DISCUSSION TITLE
Has anyone successfully added EufyCam 2 to Scrypted?

### DISCUSSION THREAD

**Discussion by JimboJamez** [koush/scrypted#1183] [Q&A]
Hi everyone

I am trying to integrate my EufyCam 2's into scrypted but no matter how I structure my rtsp stream I am getting "401 Unauthorized" and "Stream not found" errors. I have ensured that I am using the correct username and password but still keep getting the errors,

The RTSP Stream Link taken from the Eufy Security app is:

> _**rtsp://(USERNAME):(PASSWORD)@(IP ADDRESS)/live2**_

This differs from the link expected in the RTSP Stream URL in Scypted:

> _**rtsp://(IP ADDRESS)[:554]/channel/101**_ which I imagine in my circumstance should be > _**rtsp://(IP ADDRESS)/live2**_, but I stand to be corrected!

Examples below:

**rtsp://(IP ADDRESS)/live2:**

> [Rebroadcast Plugin]: Stream 1 prebuffer session started
> [Rebroadcast Plugin]: Stream 1 prebuffer session starting
> stream url rtsp://(IP ADDRESS)/live2
> [Rebroadcast Plugin]: rtsp outgoing message
> [Rebroadcast Plugin]:  OPTIONS rtsp://(IP ADDRESS)/live2 RTSP/1.0
> [Rebroadcast Plugin]: CSeq: 0
> [Rebroadcast Plugin]: User-Agent: Scrypted
> [Rebroadcast Plugin]: 
> [Rebroadcast Plugin]: rtsp incoming message
> [Rebroadcast Plugin]:  RTSP/1.0 200 OK
> [Rebroadcast Plugin]: CSeq: 0
> [Rebroadcast Plugin]: Public: OPTIONS, DESCRIBE, SETUP, TEARDOWN, PLAY, GET_PARAMETER
> [Rebroadcast Plugin]: 
> [Rebroadcast Plugin]: rtsp outgoing message
> [Rebroadcast Plugin]:  DESCRIBE rtsp://(IP ADDRESS)/live2 RTSP/1.0
> [Rebroadcast Plugin]: Accept: application/sdp
> [Rebroadcast Plugin]: CSeq: 1
> [Rebroadcast Plugin]: User-Agent: Scrypted
> [Rebroadcast Plugin]: 
> [Rebroadcast Plugin]: rtsp incoming message
> [Rebroadcast Plugin]:  RTSP/1.0 **401 Unauthorized**
> [Rebroadcast Plugin]: CSeq: 1
> [Rebroadcast Plugin]: WWW-Authenticate: Digest realm="C7H2Dpn", nonce="cuD0Cc652Hyamsa4cw31iCb"
> [Rebroadcast Plugin]: 
> [Rebroadcast Plugin]: rtsp outgoing message
> [Rebroadcast Plugin]:  DESCRIBE rtsp://(IP ADDRESS)/live2 RTSP/1.0
> [Rebroadcast Plugin]: Accept: application/sdp
> [Rebroadcast Plugin]: CSeq: 2
> [Rebroadcast Plugin]: User-Agent: Scrypted
> [Rebroadcast Plugin]: Authorization: Digest username="(USERNAME)", realm="C7H2Dpn", nonce="cuD0Cc652Hyamsa4cw31iCb", uri="rtsp://(IP ADDRESS)/live2", algorithm="MD5", response="be03e775ca4145383d24e588e7e0e6c5"
> [Rebroadcast Plugin]: 
> [Rebroadcast Plugin]: rtsp incoming message
> [Rebroadcast Plugin]:  RTSP/1.0 **404 Stream Not Found**
> [Rebroadcast Plugin]: CSeq: 2
> [Rebroadcast Plugin]: 
> [Rebroadcast Plugin]: rtsp outgoing message
> [Rebroadcast Plugin]:  TEARDOWN rtsp://(IP ADDRESS)/live2 RTSP/1.0
> [Rebroadcast Plugin]: CSeq: 3
> [Rebroadcast Plugin]: User-Agent: Scrypted
> [Rebroadcast Plugin]: Authorization: Digest username="(USERNAME)", realm="C7H2Dpn", nonce="cuD0Cc652Hyamsa4cw31iCb", uri="rtsp://(IP ADDRESS)/live2", algorithm="MD5", response="88190f0d5ed78cdb93c9f481b1d7d4b7"
> [Rebroadcast Plugin]: 
> [Rebroadcast Plugin]: Stream 1 prebuffer session ended with error M [Error]: RTSP Error: RTSP/1.0 **404 Stream Not Found**
> [Rebroadcast Plugin]:     at t.RtspClient.request (/@scrypted/prebuffer-mixin/main.nodejs.js:1:91847)
> [Rebroadcast Plugin]:     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
> [Rebroadcast Plugin]:     at async t.startRtspSession (/@scrypted/prebuffer-mixin/main.nodejs.js:1:144136)
> [Rebroadcast Plugin]:     at async V.startPrebufferSession (/@scrypted/prebuffer-mixin/main.nodejs.js:1:117624) {
> [Rebroadcast Plugin]:   status: {
> [Rebroadcast Plugin]:     line: 'RTSP/1.0 **404 Stream Not Found**',
> [Rebroadcast Plugin]:     code: 404,
> [Rebroadcast Plugin]:     version: 'RTSP/1.0',
> [Rebroadcast Plugin]:     reason: 'Stream'
> [Rebroadcast Plugin]:   }
> [Rebroadcast Plugin]: }
> [Rebroadcast Plugin]: Front Camera restarting prebuffer session in 5 seconds
> [Rebroadcast Plugin]: client error Error: read ECONNRESET
> [Rebroadcast Plugin]:     at TCP.onStreamRead (node:internal/stream_base_commons:217:20) {
> [Rebroadcast Plugin]:   errno: -104,
> [Rebroadcast Plugin]:   code: 'ECONNRESET',
> [Rebroadcast Plugin]:   syscall: 'read'
> [Rebroadcast Plugin]: }
> [Rebroadcast Plugin]: rebroadcast error Error: read ECONNRESET
> [Rebroadcast Plugin]:     at TCP.onStreamRead (node:internal/stream_base_commons:217:20) {
> [Rebroadcast Plugin]:   errno: -104,
> [Rebroadcast Plugin]:   code: 'ECONNRESET',
> [Rebroadcast Plugin]:   syscall: 'read'
> [Rebroadcast Plugin]: }

**rtsp://(IP ADDRESS):554/live2:**

> [Rebroadcast Plugin]: Stream 1 prebuffer session started
> [Rebroadcast Plugin]: Stream 1 prebuffer session starting
> stream url rtsp://(IP ADDRESS):554/live2
> [Rebroadcast Plugin]: rtsp outgoing message
> [Rebroadcast Plugin]:  OPTIONS rtsp://(IP ADDRESS):554/live2 RTSP/1.0
> [Rebroadcast Plugin]: CSeq: 0
> [Rebroadcast Plugin]: User-Agent: Scrypted
> [Rebroadcast Plugin]: 
> [Rebroadcast Plugin]: rtsp incoming message
> [Rebroadcast Plugin]:  RTSP/1.0 200 OK
> [Rebroadcast Plugin]: CSeq: 0
> [Rebroadcast Plugin]: Public: OPTIONS, DESCRIBE, SETUP, TEARDOWN, PLAY, GET_PARAMETER
> [Rebroadcast Plugin]: 
> [Rebroadcast Plugin]: rtsp outgoing message
> [Rebroadcast Plugin]:  DESCRIBE rtsp://(IP ADDRESS):554/live2 RTSP/1.0
> [Rebroadcast Plugin]: Accept: application/sdp
> [Rebroadcast Plugin]: CSeq: 1
> [Rebroadcast Plugin]: User-Agent: Scrypted
> [Rebroadcast Plugin]: 
> [Rebroadcast Plugin]: rtsp incoming message
> [Rebroadcast Plugin]:  RTSP/1.0 **401 Unauthorized**
> [Rebroadcast Plugin]: CSeq: 1
> [Rebroadcast Plugin]: WWW-Authenticate: Digest realm="uh6h2zo", nonce="AJ5GAEmgHwAIxtt3lw4523i"
> [Rebroadcast Plugin]: 
> [Rebroadcast Plugin]: rtsp outgoing message
> [Rebroadcast Plugin]:  DESCRIBE rtsp://(IP ADDRESS):554/live2 RTSP/1.0
> [Rebroadcast Plugin]: Accept: application/sdp
> [Rebroadcast Plugin]: CSeq: 2
> [Rebroadcast Plugin]: User-Agent: Scrypted
> [Rebroadcast Plugin]: Authorization: Digest username="(USERNAME)", realm="uh6h2zo", nonce="AJ5GAEmgHwAIxtt3lw4523i", uri="rtsp://(IP ADDRESS):554/live2", algorithm="MD5", response="1c325c8d25a141653702b668cc1624fd"
> [Rebroadcast Plugin]: 
> [Rebroadcast Plugin]: rtsp incoming message
> [Rebroadcast Plugin]:  RTSP/1.0 **404 Stream Not Found**
> [Rebroadcast Plugin]: CSeq: 2
> [Rebroadcast Plugin]: 
> [Rebroadcast Plugin]: rtsp outgoing message
> [Rebroadcast Plugin]:  TEARDOWN rtsp://(IP ADDRESS):554/live2 RTSP/1.0
> [Rebroadcast Plugin]: CSeq: 3
> [Rebroadcast Plugin]: User-Agent: Scrypted
> [Rebroadcast Plugin]: Authorization: Digest username="(USERNAME)", realm="uh6h2zo", nonce="AJ5GAEmgHwAIxtt3lw4523i", uri="rtsp://(IP ADDRESS):554/live2", algorithm="MD5", response="277f94296bce8d9479f3d42fe3b2a1c7"
> [Rebroadcast Plugin]: 
> [Rebroadcast Plugin]: Stream 1 prebuffer session ended with error M [Error]: RTSP Error: RTSP/1.0 **404 Stream Not Found**
> [Rebroadcast Plugin]:     at t.RtspClient.request (/@scrypted/prebuffer-mixin/main.nodejs.js:1:91847)
> [Rebroadcast Plugin]:     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
> [Rebroadcast Plugin]:     at async t.startRtspSession (/@scrypted/prebuffer-mixin/main.nodejs.js:1:144136)
> [Rebroadcast Plugin]:     at async V.startPrebufferSession (/@scrypted/prebuffer-mixin/main.nodejs.js:1:117624) {
> [Rebroadcast Plugin]:   status: {
> [Rebroadcast Plugin]:     line: 'RTSP/1.0 **404 Stream Not Found**',
> [Rebroadcast Plugin]:     code: 404,
> [Rebroadcast Plugin]:     version: 'RTSP/1.0',
> [Rebroadcast Plugin]:     reason: 'Stream'
> [Rebroadcast Plugin]:   }
> [Rebroadcast Plugin]: }
> [Rebroadcast Plugin]: Front Camera restarting prebuffer session in 5 seconds
> [Rebroadcast Plugin]: client error Error: read ECONNRESET
> [Rebroadcast Plugin]:     at TCP.onStreamRead (node:internal/stream_base_commons:217:20) {
> [Rebroadcast Plugin]:   errno: -104,
> [Rebroadcast Plugin]:   code: 'ECONNRESET',
> [Rebroadcast Plugin]:   syscall: 'read'
> [Rebroadcast Plugin]: }
> [Rebroadcast Plugin]: rebroadcast error Error: read ECONNRESET
> [Rebroadcast Plugin]:     at TCP.onStreamRead (node:internal/stream_base_commons:217:20) {
> [Rebroadcast Plugin]:   errno: -104,
> [Rebroadcast Plugin]:   code: 'ECONNRESET',
> [Rebroadcast Plugin]:   syscall: 'read'
> [Rebroadcast Plugin]: }

**rtsp://(USERNAME):(PASSWORD)@(IP ADDRESS):554/live2:**

> [Rebroadcast Plugin]: Stream 1 prebuffer session started
> [Rebroadcast Plugin]: Stream 1 prebuffer session starting
> stream url rtsp://(USERNAME):(PASSWORD)@(IP ADDRESS):554/live2
> [Rebroadcast Plugin]: rtsp outgoing message
> [Rebroadcast Plugin]:  OPTIONS rtsp://(IP ADDRESS):554/live2 RTSP/1.0
> [Rebroadcast Plugin]: CSeq: 0
> [Rebroadcast Plugin]: User-Agent: Scrypted
> [Rebroadcast Plugin]: 
> [Rebroadcast Plugin]: rtsp incoming message
> [Rebroadcast Plugin]:  RTSP/1.0 200 OK
> [Rebroadcast Plugin]: CSeq: 0
> [Rebroadcast Plugin]: Public: OPTIONS, DESCRIBE, SETUP, TEARDOWN, PLAY, GET_PARAMETER
> [Rebroadcast Plugin]: 
> [Rebroadcast Plugin]: rtsp outgoing message
> [Rebroadcast Plugin]:  DESCRIBE rtsp://(IP ADDRESS):554/live2 RTSP/1.0
> [Rebroadcast Plugin]: Accept: application/sdp
> [Rebroadcast Plugin]: CSeq: 1
> [Rebroadcast Plugin]: User-Agent: Scrypted
> [Rebroadcast Plugin]: 
> [Rebroadcast Plugin]: rtsp incoming message
> [Rebroadcast Plugin]:  RTSP/1.0 **401 Unauthorized**
> [Rebroadcast Plugin]: CSeq: 1
> [Rebroadcast Plugin]: WWW-Authenticate: Digest realm="0x0c2D9", nonce="CAiej6pawBlbttkD3obbfj2"
> [Rebroadcast Plugin]: 
> [Rebroadcast Plugin]: rtsp outgoing message
> [Rebroadcast Plugin]:  DESCRIBE rtsp://(IP ADDRESS):554/live2 RTSP/1.0
> [Rebroadcast Plugin]: Accept: application/sdp
> [Rebroadcast Plugin]: CSeq: 2
> [Rebroadcast Plugin]: User-Agent: Scrypted
> [Rebroadcast Plugin]: Authorization: Digest username="(USERNAME)", realm="0x0c2D9", nonce="CAiej6pawBlbttkD3obbfj2", uri="rtsp://(IP ADDRESS):554/live2", algorithm="MD5", response="d84ccc9c4a733c6d3bebf3c0b6e97eeb"
> [Rebroadcast Plugin]: 
> [Rebroadcast Plugin]: rtsp incoming message
> [Rebroadcast Plugin]:  RTSP/1.0 **404 Stream Not Found**
> [Rebroadcast Plugin]: CSeq: 2
> [Rebroadcast Plugin]: 
> [Rebroadcast Plugin]: rtsp outgoing message
> [Rebroadcast Plugin]:  TEARDOWN rtsp://(IP ADDRESS):554/live2 RTSP/1.0
> [Rebroadcast Plugin]: CSeq: 3
> [Rebroadcast Plugin]: User-Agent: Scrypted
> [Rebroadcast Plugin]: Authorization: Digest username="(USERNAME)", realm="0x0c2D9", nonce="CAiej6pawBlbttkD3obbfj2", uri="rtsp://(IP ADDRESS):554/live2", algorithm="MD5", response="5bf829aa5a75c474a03e7541c344c67e"
> [Rebroadcast Plugin]: 
> [Rebroadcast Plugin]: Stream 1 prebuffer session ended with error M [Error]: RTSP Error: RTSP/1.0 **404 Stream Not Found**
> [Rebroadcast Plugin]:     at t.RtspClient.request (/@scrypted/prebuffer-mixin/main.nodejs.js:1:91847)
> [Rebroadcast Plugin]:     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
> [Rebroadcast Plugin]:     at async t.startRtspSession (/@scrypted/prebuffer-mixin/main.nodejs.js:1:144136)
> [Rebroadcast Plugin]:     at async V.startPrebufferSession (/@scrypted/prebuffer-mixin/main.nodejs.js:1:117624) {
> [Rebroadcast Plugin]:   status: {
> [Rebroadcast Plugin]:     line: 'RTSP/1.0 **404 Stream Not Found**',
> [Rebroadcast Plugin]:     code: 404,
> [Rebroadcast Plugin]:     version: 'RTSP/1.0',
> [Rebroadcast Plugin]:     reason: 'Stream'
> [Rebroadcast Plugin]:   }
> [Rebroadcast Plugin]: }
> [Rebroadcast Plugin]: Front Camera restarting prebuffer session in 5 seconds
> [Rebroadcast Plugin]: client error Error: read ECONNRESET
> [Rebroadcast Plugin]:     at TCP.onStreamRead (node:internal/stream_base_commons:217:20) {
> [Rebroadcast Plugin]:   errno: -104,
> [Rebroadcast Plugin]:   code: 'ECONNRESET',
> [Rebroadcast Plugin]:   syscall: 'read'
> [Rebroadcast Plugin]: }
> [Rebroadcast Plugin]: rebroadcast error Error: read ECONNRESET
> [Rebroadcast Plugin]:     at TCP.onStreamRead (node:internal/stream_base_commons:217:20) {
> [Rebroadcast Plugin]:   errno: -104,
> [Rebroadcast Plugin]:   code: 'ECONNRESET',
> [Rebroadcast Plugin]:   syscall: 'read'
> [Rebroadcast Plugin]: }

Has anyone been able to successfully do this, and if so are you able to share your secrets please?

**Comment by jackoncept** [comment]
Did you manage to find a solution after several months ?

**Comment by pieterdevriesch** [comment]
I've just succesfully added it by setting the authentication to basic in the Eufy app settings, and adding the :554 to the url that it gives you just to be sure.

**Comment by Flammy004** [comment]
I had to set it to continous recording in the Eufy app to get it working. :)
